app/actions/whitelistedDApps.js

```diff
@@ -0,0 +1,12 @@
+export const ADD_WHITELISTED_DAPP = 'ADD_WHITELISTED_DAPP'
```

Comment:
Can you explain me briefly how is the architecture of the React App? does it have only one "view"? Which are the main parts of the app in terms of user interfaces?

Reply:
After setting a password, the user can create a random account or restore an account from a phrase and  encrypt this account with the password. After that, the account view is shown. The user can also add or delete some URLs to whitelist the dapps in order to inject the web3 provider.